src/AdminService.Business/Commands/InstallAppCommand.cs

```diff
@@ -0,0 +1,166 @@
+using Microsoft.Extensions.Logging;
+
+namespace LT.DigitalOffice.AdminService.Business.Commands
+{
+  public class InstallAppCommand
+  {
+    private readonly IAppInstallationMapper _mapper;
+    private readonly ILogger<IInstallAppCommand> _logger;
+    private readonly IInstallAppRequestValidator _validator;
+    private readonly IServiceConfigurationRepository _repository;
+    private readonly IRequestClient<ICreateAdminRequest> _rcCreateAdmin;
+    private readonly IRequestClient<ICreateSmtpCredentialsRequest> _rcCreateSmtp;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public InstallAppCommand(
+      IAppInstallationMapper mapper,
+      ILogger<IInstallAppCommand> logger,
+      IInstallAppRequestValidator validator,
+      IServiceConfigurationRepository repository,
+      IRequestClient<ICreateAdminRequest> rcCreateAdmin,
+      IRequestClient<ICreateSmtpCredentialsRequest> rcCreateSmtp,
+      IHttpContextAccessor httpContextAccessor)
+    {
+      _mapper = mapper;
+      _logger = logger;
+      _validator = validator;
+      _repository = repository;
+      _rcCreateAdmin = rcCreateAdmin;
+      _rcCreateSmtp = rcCreateSmtp;
+      _httpContextAccessor = httpContextAccessor;
+    }
+
+    private async Task<bool> CreateSmtp(SmtpInfo smtpInfo, List<string> errors)
+    {
+      string message = "Can not create smtp credentials.";
+
+      try
+      {
+        Response<IOperationResult<bool>> response = await _rcCreateSmtp.GetResponse<IOperationResult<bool>>(
+          ICreateSmtpCredentialsRequest.CreateObj(
+            host: smtpInfo.Host,
+            port: smtpInfo.Port,
+            enableSsl: smtpInfo.EnableSsl,
+            email: smtpInfo.Email,
+            password: smtpInfo.Password));
+
+        if (response.Message.IsSuccess && response.Message.Body)
+        {
+          return true;
+        }
+
+        _logger.LogWarning(message, string.Join("\n", response.Message.Errors));
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, message);
+      }
+      errors.Add(message);
+
+      return false;
+    }
+
+    private async Task<bool> CreateAdmin(AdminInfo info, List<string> errors)
+    {
+      string message = "Can not create admin.";
+
+      try
+      {
+        Response<IOperationResult<bool>> response = await _rcCreateAdmin.GetResponse<IOperationResult<bool>>(
+          ICreateAdminRequest.CreateObj(
+            info.FirstName, 
+            info.MiddleName, 
+            info.LastName, 
+            info.Email, 
+            info.Login, 
+            info.Password));
+
+        if (response.Message.IsSuccess && response.Message.Body)
+        {
+          return true;
+        }
+
+        errors.Add(message);
+
+        _logger.LogWarning(message, string.Join("\n", response.Message.Errors));
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, message);
+
+        errors.Add(message);
+      }
+

```

Comment:
you can write errors.Add(message); here and remove it from 102 and 110 line

Reply:
fixed